subsys/net/ip/net_shell.c

```diff
@@ -4768,6 +5172,8 @@ SHELL_STATIC_SUBCMD_SET_CREATE(net_commands,
 		  cmd_net_conn),
 	SHELL_CMD(dns, &net_cmd_dns, "Show how DNS is configured.",
 		  cmd_net_dns),
+	SHELL_CMD(events, &net_cmd_events, "Monitor network management events.",
```

Comment:
This can be the conditional command:

```
SHELL_COND_CMD(CONFIG_NET_MGMT_EVENT_MONITOR_AUTO_START,
 events, &net_cmd_events, "Monitor network management events.", cmd_net_events),
```
In such case you will not need to get uart pointer.

Reply:
Hmm, the `CONFIG_NET_MGMT_EVENT_MONITOR_AUTO_START` is there just to tell that the monitoring should be started when `net_shell_init()` is called (like in the beginning of the device startup). The purpose here is not to disable the "net events" command.